zwave_js/rootfs/etc/cont-init.d/config.sh

```diff
@@ -4,32 +4,62 @@
 # ==============================================================================
 declare network_key
 
+flush_to_disk=false
+
 readonly DOCS_EXAMPLE_KEY="2232666D100F795E5BB17F0A1BB7A146"
 
-if [[ "${DOCS_EXAMPLE_KEY}" == "$(bashio::config 'network_key')" ]]; then
-    bashio::log.fatal
-    bashio::log.fatal 'The add-on detected that the Z-Wave network key used'
-    bashio::log.fatal 'is from the documented example.'
-    bashio::log.fatal
-    bashio::log.fatal 'Using this key is insecure, because it is publicly'
-    bashio::log.fatal 'listed in the documentation.'
-    bashio::log.fatal
-    bashio::log.fatal 'Please check the add-on documentation on how to'
-    bashio::log.fatal 'create your own, secret, "network_key" and replace'
-    bashio::log.fatal 'the one you have configured.'
-    bashio::log.fatal
-    bashio::log.fatal 'Click on the "Documentation" tab in the Z-Wave JS'
-    bashio::log.fatal 'add-on panel for more information.'
-    bashio::log.fatal
-    bashio::exit.nok
-elif ! bashio::config.has_value 'network_key'; then
-    bashio::log.info "No Network Key is set, generate one..."
-    network_key=$(hexdump -n 16 -e '4/4 "%08X" 1 "\n"' /dev/random)
-    bashio::addon.option network_key "${network_key}"
-else
-    network_key=$(bashio::config 'network_key')
+if bashio::config.has_value 'network_key'; then
+    bashio::log.info "Migrating \"network_key\" option to \"s0_legacy_key\"..."
+    network_key=$(bashio::string.upper "$(bashio::config 'network_key')")
+    bashio::addon.option s0_legacy_key "${network_key}"
+    bashio::addon.option network_key
+    flush_to_disk=true
+fi
+
+# We need to restart if we migrated the key so it gets flushed to disk
+if [[ ${flush_to_disk} ]]; then
+    bashio::log.info "Flushing config to disk due to migration"
+    bashio::addon.options > "/data/options.json"
+    flush_to_disk=false
```

Comment:
There is only 1 case, so no need to have an extra construct for that in this case

```suggestion
    bashio::log.info "Flushing config to disk due to migration"
    bashio::addon.options > "/data/options.json"
```